pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java

```diff
@@ -320,6 +331,14 @@ public ByteBuffer getState(String key) {
 
     @Override
     public void recordMetric(String metricName, double value) {
+        userMetrics.computeIfAbsent(metricName,
+                s -> Gauge.build()
+                .name("pulsar_function_user_metric_" + metricName)
+                        .help("Pulsar Function user metric " + metricName + ".")
+                .labelNames(FunctionStats.metricsLabelNames)
+                .register(collectorRegistry));
+
+        userMetrics.get(metricName).labels(metricsLabels).set(value);
         currentAccumulatedMetrics.putIfAbsent(metricName, new AccumulatedMetricDatum());
```

Comment:
what's the use of currentAccumulatedMetrics now?

Reply:
we can get rid of it but that is another discussion on how the user API for metrics is for functions

Reply:
yup get rid of currentAccumulatedMetrics